### PR TITLE
Log post tests

### PR DESCRIPTION
**Pull request by henridf**

### CODE REVIEW

File: zqd/handlers_test.go [hunk 1]
```diff
@@ -313,8 +322,69 @@ func postSpaceData(t *testing.T, c *api.Connection, spaceName, data string) *api
...
 		payloads = append(payloads, p)
 	}
-	last := payloads[len(payloads)-1].(*api.TaskEnd)
-	assert.Equal(t, last.Type, "TaskEnd")
-	assert.Nil(t, last.Error)
-	return s
+	return payloads
+}
+
+func TestPostLogs(t *testing.T) {
```
Comment (mattnibs) [inline zqd/handlers_test.go:328]: I've been keeping all the `Test` functions above the utility functions to maintain readability. Are you cool with sticking to this convention?

File: zqd/handlers_test.go [hunk 2]
```diff
@@ -313,8 +322,69 @@ func postSpaceData(t *testing.T, c *api.Connection, spaceName, data string) *api
...
+		"0:[conn;2;CBrzd94qfowOqJwCHa;]",
+	}
+	_, client, done := newCore(t)
+	defer done()
+	spaceName := "test"
+	t.Run("Space", func(t *testing.T) {
+		_, err := client.SpacePost(context.Background(), api.SpacePostRequest{Name: spaceName})
+		require.NoError(t, err)
+	})
+
```
Comment (mattnibs) [inline zqd/handlers_test.go:344]:
```suggestion
```

File: zqd/handlers_test.go [hunk 3]
```diff
@@ -290,19 +291,27 @@ func newCoreAtDir(t *testing.T, dir string) (*zqd.Core, *api.Connection, func())
...
 	defer f.Close()
-	defer os.Remove(name)
-	_, err = f.WriteString(data)
+	_, err = f.WriteString(contents)
 	require.NoError(t, err)
+	return f.Name()
+}
+
+// postSpaceLogs POSTs the provided strings as logs in to the provided space, and returns a slice of any payloads that the server sent.
+func postSpaceLogs(t *testing.T, c *api.Connection, spaceName string, logs []string) []interface{} {
```
Comment (mattnibs) [inline zqd/handlers_test.go:305]: Can we instead use the `...` syntax for logs:
```suggestion
func postSpaceLogs(t *testing.T, c *api.Connection, spaceName string, logs ...string) []interface{} {
```

File: zqd/handlers_test.go (hunk 1 above)
Comment (henridf) [reply zqd/handlers_test.go]: This doesn't affect readability either way for me, but I've changed it.

File: zqd/handlers_test.go (hunk 3 above)
Comment (henridf) [reply zqd/handlers_test.go]: 👍

**Pull request merged**